modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+    expect(currentUserId).to.equal('test-user-1', 'Should return the configured user ID');
+  });
+
+  it('TINY-11974: Should have frozen getUserId but mutable fetchUsers', () => {
+    const editor = hook.editor();
+    const lookup = editor.userLookup;
+
+    try {
+      lookup.getUserId = Fun.constant('new-id');
+      expect.fail('Should not allow modifying getUserId');
+    } catch (error) {
+      expect(error).to.be.instanceOf(Error, 'Should throw error when trying to modify read-only function');
+    }
+
+    expect(lookup.getUserId()).to.equal('test-user-1', 'Should maintain original value');
+  });
+
+  it('TINY-11974: Should fetch users and return array of promises', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.be.an('array', 'Should return an array of promises');
+    expect(promises).to.have.lengthOf(1, 'Should return array with one promise');
+    expect(promises[0]).to.be.instanceOf(Promise, 'Should return a Promise');
+  });
+
+  it('TINY-11974: Should return multiple promises for multiple userIds', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1', 'test-user-2' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.have.lengthOf(2, 'Should return array with two promises');
+    expect(promises[0]).to.be.instanceOf(Promise, 'First item should be a Promise');
+    expect(promises[1]).to.be.instanceOf(Promise, 'Second item should be a Promise');
+  });
+
+  it('TINY-11974: Should throw an error for empty string ids', async () => {
+    const editor = hook.editor();
+    const userIds = [ '' ];
+    const [ userPromise ] = editor.userLookup.fetchUsers(userIds);
+
+    await expect(userPromise).to.eventually.be.rejectedWith(
+      Error,
+      'User  not found',
+      'Should throw error for empty string ID'
+    );
+  });
+
+  it('TINY-11974: Should resolve with user data when fetching a single user', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    const [ userPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+    const expectedUser = {
+      id: userId,
+      name: 'Test User',
+      avatar: 'test-avatar.png',
+      description: 'Test Description'
+    };
+
+    await expect(userPromise).to.eventually.deep.equal(
+      expectedUser,
+      'Should resolve with expected user object'
+    );
+  });
+
+  it('TINY-11974: Should cache and return same data for subsequent requests', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    const [ firstPromise ] = editor.userLookup.fetchUsers([ userId ]);
+    const [ secondPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+    await expect(firstPromise).to.eventually.deep.equal(
+      await secondPromise,
+      'Should return same data for subsequent requests'
+    );
```

Comment:
You don't need deep equal. This just asserts the _contents_ are the same, which it would be even if they were not cached.

The check should be just `firstPromise === secondPromise` (or the chai equivalent).